golinks.go

```diff
@@ -1,156 +1,160 @@
 package main
 
 import (
-	"encoding/json"
-	"flag"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"regexp"
-	"strings"
+        "encoding/json"
+        "flag"
+        "fmt"
+        "io/ioutil"
+        "log"
+        "net/http"
+        "regexp"
+        "strings"
 )
 
-type Redirect struct {
-	Shortname string
-	Url       string
-	Requests  int32
+type redirect struct {
+        shortname string
+        url       string
+        requests  int32
 }
 
-type Settings struct {
-	Redirects []*Redirect
-	Filename  string
+type settings struct {
+        redirects []*redirect
+        filename  string
 }
 
 func main() {
-	port := flag.String("http_port", "8080", "Port number to listen")
-	configFile := flag.String("config", "redirects.json", "Port number to listen")
-	flag.Parse()
-	log.Printf("starting golinks...")
-	r := NewRedirector(*configFile)
-	r.ReadConfig()
-
-	http.HandleFunc("/add/", r.AddLink)
-	http.HandleFunc("/list/", r.GetLinks)
-	http.HandleFunc("/del/", r.DelLink)
-	http.HandleFunc("/", r.Redirect)
-	if err := http.ListenAndServe(":"+*port, nil); err != nil {
-		log.Fatal("Unable to listen %s", err)
-	}
-	for {
-	}
+        port := flag.String("http_port", "8080", "Port number to listen")
+        configFile := flag.String("config", "redirects.json", "Configuration filename")
+        flag.Parse()
+        log.Printf("Starting golinks...")
+        r := newRedirector(*configFile)
+        r.readConfig()
+
+        http.HandleFunc("/add/", r.addLink)
+        http.HandleFunc("/list/", r.getLinks)
+        http.HandleFunc("/del/", r.delLink)
+        http.HandleFunc("/", r.redirect)
+        if err := http.ListenAndServe(":"+*port, nil); err != nil {
+                log.Fatal("Unable to listen %s", err)
+        }
 }
 
-func NewRedirector(file string) *Settings {
-	return &Settings{
-		Filename: file,
-	}
+func newRedirector(file string) *settings {
+        return &settings{
+                filename: file,
+        }
 }
 
-func (m *Settings) ReadConfig() {
-	log.Printf("reading configuration...")
-	jsonBlob, err := ioutil.ReadFile(m.Filename)
-	if err != nil {
-		log.Printf("Unable to read configuration %s", err)
```

Comment:
this error message confused me when i ran it the first time. how about something like "no config file found.. creating new config"